packages/api/endpoints/reconciliation-reports.js

```diff
@@ -77,35 +88,41 @@ async function getReport(req, res) {
     );
 
     if (Key.endsWith('.json') || Key.endsWith('.csv')) {
-      const reportSize = await getObjectSize({ s3: s3(), bucket: Bucket, key: Key });
+      const reportSize = await getObjectSize({ s3: s3(), bucket: Bucket, key: Key }) ?? 0;
       // estimated payload size, add extra
       const estimatedPayloadSize = presignedS3Url.length + reportSize + 50;
-      if (
-        estimatedPayloadSize
-          > (process.env.maxResponsePayloadSizeBytes || maxResponsePayloadSizeBytes)
-      ) {
+      let maxResponsePayloadSize;
+      if (process.env.maxResponsePayloadSizeBytes) {
+        maxResponsePayloadSize = Number(process.env.maxResponsePayloadSizeBytes)
+      } else {
+        maxResponsePayloadSize = maxResponsePayloadSizeBytes
+      }
+      if (estimatedPayloadSize > maxResponsePayloadSize) {
         res.json({
           presignedS3Url,
           data: `Error: Report ${name} exceeded maximum allowed payload size`,
         });
       } else {
         const file = await getS3Object(Bucket, Key);
```

Comment:
there's a warning that `getS3Object` is a deprecated method, and if you look at where it's defined, we actually explicitly marked it as deprecated. So I'm unsure if there's a different method we should be using here, there wasn't a clear 'replacement' method to use.